test/e2e/grafana/generate_test.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grafana
+
+import (
+	"path/filepath"
+
+	pluginutil "sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/kubebuilder/v3/test/e2e/utils"
+)
+
+var _ = Describe("kubebuilder", func() {
+	Context("plugin grafana/v1-alpha", func() {
+		var (
+			kbc *utils.TestContext
+		)
+
+		BeforeEach(func() {
+			var err error
+			kbc, err = utils.NewPureKbTester("GO111MODULE=on")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(kbc.Prepare()).To(Succeed())
+		})
+
+		AfterEach(func() {
+			kbc.Destroy()
+		})
+
+		It("should generate a runnable project with grafana plugin", func() {
+			GenerateProject(kbc)
+		})
+
+	})
+})
+
+// GenerateProject implements a grafana/v1(-alpha) plugin project defined by a TestContext.
+func GenerateProject(kbc *utils.TestContext) {
+	var err error
+
+	By("initializing a project")
+	err = kbc.Init(
+		"--plugins", "grafana.kubebuilder.io/v1-alpha",
+	)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	By("uncomment grafana/custom-metrics/config.yaml to verify config template")
+	ExpectWithOffset(1, pluginutil.UncommentCode(
+		filepath.Join(kbc.Dir, "grafana", "custom-metrics", "config.yaml"),
+		"#  - metric: # Raw custom metric (required)", "#")).To(Succeed())
+	ExpectWithOffset(1, pluginutil.UncommentCode(
+		filepath.Join(kbc.Dir, "grafana", "custom-metrics", "config.yaml"),
+		"#    type:   # Metric type: counter/gauge/histogram (required)", "#")).To(Succeed())
+
+	ExpectWithOffset(1, pluginutil.UncommentCode(
```

Comment:
If we are uncommented and then replace the file we do not need to uncomment.
IMO here we can change that to use ReplaceInFile and you will pass a const which has the original content to be replaced for new content, see an example: https://github.com/kubernetes-sigs/kubebuilder/blob/20828c7e87e36c005f77e9c57ed9a84371dc65d8/test/e2e/v3/generate_test.go#L345-L359